XMonad/Prompt.hs

```diff
@@ -1,11 +1,12 @@
-{-# LANGUAGE ScopedTypeVariables #-}
-{-# LANGUAGE TupleSections #-}
-{-# LANGUAGE FlexibleContexts #-}
-{-# LANGUAGE ExistentialQuantification #-}
 {-# LANGUAGE CPP #-}
+{-# LANGUAGE ExistentialQuantification #-}
```

Comment:
I'd be careful about alpha-sorting these given that ghc can add or remove pragmas based on other pragmas. In other words, order can matter.

Reply:
Can it really? I don't think adding pragmas would cause any problems, but removing sounds problematic indeed. I've never heard that language extensions have these capabilities, though, do you have an example?

Reply:
The GHC manual describes this as "implies". See for example [Type families](https://downloads.haskell.org/ghc/9.4.7/docs/users_guide/exts/type_families.html#type-families), which says:

Implies: MonoLocalBinds, KindSignatures, ExplicitNamespaces

Off the top of my head I can't think of any that get turned off by other extensions, but it's certainly possible so some care needs to be given to ordering as an extension may thereby get ignored or conversely might lead to odd results if one normally turned off gets turned on later.

Reply:
I knew about the "implies" stuff, but that should be irrelevant of the order (and we're not turning anything off here). Anyways, I will leave the old sorting in place for now, only adding the new extension to the bottom

Reply:
Sadly, it's not independent of the order and things like turning `MonoLocalBinds` back off after enabling `TypeFamilies` are known idioms.